src/remediations/remediations.queries.js

```diff
@@ -517,6 +518,97 @@ exports.getIssueSystems = function (id, tenant_org_id, created_by, issueId) {
     });
 };
 
+// Return a paginated, sorted list of distinct systems for a remediation plan
+exports.getPlanSystems = async function (
+    remediation_plan_id,
+    tenant_org_id,
+    created_by,
+    column = 'display_name',
+    asc = true,
+    filter = null,
+    limit = 50,
+    offset = 0
+) {
+    const { Op, s: { literal, col, cast }, issue, issue_system, remediation } = db;
+
+    // Fetch distinct system ids for the plan scoped by tenant/user
+    const distinctSystemIds = (await issue_system.findAll({
+        attributes: [[db.s.fn('DISTINCT', db.s.col('issue_system.system_id')), 'system_id']],
+        include: [{
+            attributes: [],
+            model: issue,
+            required: true,
+            where: { remediation_id: remediation_plan_id },
+            include: [{
+                attributes: [],
+                association: issue.associations.remediation,
+                model: remediation,
+                required: true,
+                where: { tenant_org_id, created_by }
+            }]
+        }],
+        raw: true
+    })).map(r => r.system_id);
+
+    // If no systems found for the remediation plan, return empty results
+    if (distinctSystemIds.length === 0) {
+        return { count: 0, rows: [] };
+    }
+
+    // If there are systems that don't exist in the systems table yet,
```

Comment:
These next few lines will check to see if there are any systems in the remediation plan that don't exist in the systems table for whatever reason and if there are missing systems, we'll call Inventory and retrieve them and then store them in the database 

This hopefully won't happen often but just an edge case